### PR TITLE
Display Spectators in Scoreboard

### PR DESCRIPTION
**Pull request by VelocityRa**
For discussion see the relevant Issue: #412.

Current look:

### CODE REVIEW

File: Sources/Client/ScoreboardView.cpp [hunk 1]
```diff
@@ -297,5 +313,73 @@ namespace spades {
...
+
+				numSpectators++;
+
+				// Measure total width in pixels so that we can center align all the spectators
+				sprintf(buf, "#%d", ent.id);
+				totalPixelWidth += font->Measure(buf).x + font->Measure(ent.name).x + xPixelSpectatorOffset;
+			}
+			switch (numSpectators) {
+				case 0: return;
+				case 1:	strcpy(buf, "Spectator:");
```
Comment (theunknownxy) [inline Sources/Client/ScoreboardView.cpp]: I think you can use [GetTextPlural](https://github.com/yvt/openspades/blob/master/Sources/Core/Strings.h#L267) to make it more localization friendly.

Comment (VelocityRa) [inline Sources/Client/ScoreboardView.cpp]: Good catch, I'll fix that, thanks.

Comment (feikname) [inline Sources/Client/ScoreboardView.cpp]: Was about to say that

Comment (VelocityRa) [inline Sources/Client/ScoreboardView.cpp]: "Spectator" doesn't belong in the localization lists though, it's not translated in any language.

Comment (theunknownxy) [inline Sources/Client/ScoreboardView.cpp]: Not yet?

Comment (VelocityRa) [inline Sources/Client/ScoreboardView.cpp]: Ah, this function literally returns the plural depending on the language, right, nvm.

Comment (feikname) [inline Sources/Client/ScoreboardView.cpp]: Could you please tell how you did it? I was wondered :)

My guess is update-pot.sh

**Comment by VelocityRa** [comment]
There we go fixed.
Having second thoughts about that colon now... It looks fine but it's not consistent with the the teams' texts.

**Comment by feikname** [comment]
I think it looks nice with the colon.

**Comment by feikname** [comment]
@VelocityRa there is something wrong with your implementation of plural forms, it always stays on the singular, tested it right now.
image

**Comment by VelocityRa** [comment]
@feikname Have you specified the plural forms [correctly](https://github.com/yvt/openspades/blob/master/Resources/Locales/de/openspades.po#L1573)? It should work.

**Comment by feikname** [comment]
I tried to simplify the line to `strcpy(buf, _TrN("Client", "Spectator", "Spectators", numSpectators).c_str());`, updated the `.pot` file and double-check my translation and is still not working. 

The most weird part is that without any`core_locale` it works fine.

**Comment by VelocityRa** [comment]
Based on other usages, I *think* I'm using this macro correctly.
Gonna have to ask @yvt's help on [this](https://github.com/yvt/openspades/pull/466/commits/ac35fcea0811396aae074b7d6e7bc86fe3d7c4ce).

**Comment by feikname** [comment]
I will double check myself too. I am almost 100% sure you did the right implementation, but I can't get it to work. Perhaps it's some dark auto-casting happening with NumSpectators

**Comment by VelocityRa** [comment]
Test it with `1` and `2` instead of `numSpectators`. Does it work correctly then?

**Comment by feikname** [comment]
I can only try a few hours from now, I will test as soon as I arrive back home

**Comment by feikname** [comment]
Ok, I will try now.

**Comment by feikname** [comment]
@VelocityRa status update:

Ok, I tested and uploaded my changes to [my fork](https://github.com/feikname/openspades) of your fork.

I made sure to double-check the translations and also named them them as `Singular{1}` and `Plural{1}`, so I am almost sure there's no way I wronged them. In case you wanna see for yourself, this is [the commit] where I created them](https://github.com/feikname/openspades/commit/ff81607c3d7f66ff13b566b4f3b1fc171fbcd655). 

The `.pot` was generated from `update-pot.sh` (xgettext) and the `pt_br/openspades.po` was edited from *Poedit*, so I am exluding any syntax errors too. `update-pot.sh` was run from *Bash on Windows* (lxss).

I also used "2" as `numSpectators`, so it's not a issue with `numSpectators` value.

**When my `core_locale` is set to "pt_br", this is the result:**
image

**However, with `core_locale` set to empty (nothing), it works correctly 😒**
image

----

This is very weird, maybe a new issue about this should be opened? If you can confirm on your system too, I'd be very grateful.

**Comment by VelocityRa** [comment]
>This is very weird, maybe a new issue about this should be opened? If you can confirm on your system too, I'd be very grateful.

Yeah it's present on my system too.

**Comment by VelocityRa** [comment]
@yvt Rebased your commits to test the locale thing and it works now!
Should I merge this?

**Pull request merged**